packages/alea-frontend/components/EndSemSumAccordion.tsx

```diff
@@ -0,0 +1,229 @@
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (courseId) {
+      fetchQuizzes();
+    }
+  }, [courseId]);
+
+  const handleQuizToggle = (quizId: string) => {
+    setExcludedQuizzes((prev) =>
+      prev.includes(quizId) ? prev.filter((id) => id !== quizId) : [...prev, quizId]
+    );
+  };
+
+  const handleSelectAll = () => {
+    if (excludedQuizzes.length === quizzes.length) {
+      setExcludedQuizzes([]);
+    } else {
+      setExcludedQuizzes(quizzes.map((quiz) => quiz.id));
+    }
+  };
+
+  const handleGenerateSummary = async () => {
+    try {
+      setGenerating(true);
+      setError(null);
+      setSuccess(null);
+      const result = await generateEndSemesterSummary(courseId, courseInstance, excludedQuizzes, topN);
+      if (result?.csvData) {
+        downloadFile(result.csvData, `summary_${courseId}_${courseInstance}.csv`, 'text/csv');
+        setSuccess(result.message || 'Summary generated and downloaded successfully!');
+        setTimeout(() => setSuccess(''), 3000);
+      } else {
+        throw new Error('CSV data not available');
+      }
+    } catch (err) {
+      console.error('Error generating summary:', err);
+      setError('Failed to generate end semester summary. Please try again.');
+    } finally {
+      setGenerating(false);
+    }
+  };
+
+  const formatDate = (timestamp: number) => {
+    return new Date(timestamp).toLocaleDateString('en-US', {
+      year: 'numeric',
+      month: 'short',
+      day: 'numeric',
+      hour: '2-digit',
+      minute: '2-digit',
+    });
+  };
+
+  return (
+    <Accordion>
+      <AccordionSummary
+        expandIcon={<ExpandMoreIcon />}
+        aria-controls="end-sem-summary-content"
+        id="end-sem-summary-header"
+      >
+        <Typography variant="h6">End Semester Summary</Typography>
+      </AccordionSummary>
+      <AccordionDetails>
+        <Box>
+           {error && (
+          <Alert severity="error" sx={{ mb: 2 }} onClose={() => setError('')}>
+            {error}
+          </Alert>
+        )}
+
+        {success && (
+          <Alert severity="success" sx={{ mb: 2 }} onClose={() => setSuccess('')}>
+            {success}
+          </Alert>
+        )}
+
+          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
+            Select quizzes to exclude from the end semester summary calculation:
+          </Typography>
+
+          {loading ? (
+            <Box display="flex" justifyContent="center" alignItems="center" p={2}>
+              <CircularProgress size={24} />
+              <Typography sx={{ ml: 1 }}>Loading quizzes...</Typography>
+            </Box>
+          ) : quizzes.length === 0 ? (
+            <Typography color="text.secondary">No quizzes found for this course.</Typography>
+          ) : (
+            <>
+              <Box sx={{ mb: 2 }}>
+                <Button size="small" variant="outlined" onClick={handleSelectAll} sx={{ mb: 1 }}>
```

Comment:
this will never be used. The instructor will select at most one or two quizzes to exclude. Remove all code & functionality related to "select all"